src/main/kotlin/no/skatteetaten/aurora/gradle/plugins/mutators/JavaApplicationTools.kt

```diff
@@ -0,0 +1,497 @@
+                }
+            }
+
+            with(dependencies) {
+                implementationDependencies.forEach { add("implementation", it) }
+            }
+        }
+
+        val resolvedBootJarText = when {
+            bootJarEnabled -> ", bootJar enabled"
+            else -> ", bootJar disabled"
+        }
+        val resolvedWebFluxText = when {
+            webFluxEnabled -> ", webflux enabled and webmvc + tomcat excluded"
+            else -> ", webFlux disabled"
+        }
+
+        return AuroraReport(
+            name = "plugin org.springframework.boot",
+            dependenciesAdded = implementationDependencies.map {
+                "implementation $it"
+            },
+            description = "Build info$resolvedBootJarText$resolvedWebFluxText, Optional devtools",
+            pluginsApplied = listOf("io.spring.dependency-management")
+        )
+    }
+
+    fun applyDefaultPlugins(): AuroraReport {
+        with(project) {
+            plugins.apply("java")
+            plugins.apply("maven")
+        }
+
+        return AuroraReport(
+            name = "aurora.applyDefaultPlugins",
+            pluginsApplied = listOf("java", "maven")
+        )
+    }
+
+    fun applyJavaDefaults(compability: String): AuroraReport {
+        with(project) {
+            withGroovyBuilder {
+                "sourceCompatibility" to compability
+            }
+
+            if (extensions.extraProperties.has("version")) {
+                version = extensions.extraProperties["version"] as String
+            }
+
+            if (extensions.extraProperties.has("groupId")) {
+                group = extensions.extraProperties["groupId"] as String
+            }
+        }
+
+        return AuroraReport(
+            name = "aurora.applyJavaDefaults",
+            description = "Set groupId, version and add sourceCompability"
+        )
+    }
+
+    fun applySpockSupport(
+        groovyVersion: String,
+        spockVersion: String,
+        cglibVersion: String,
+        objenesisVersion: String
+    ): AuroraReport {
+        val testDependencies = mutableListOf(
+            "org.codehaus.groovy:groovy-all:$groovyVersion",
+            "org.spockframework:spock-core:$spockVersion",
+            "cglib:cglib-nodep:$cglibVersion",
+            "org.objenesis:objenesis:$objenesisVersion"
+        )
+
+        project.plugins.withId("org.springframework.boot") {
+            testDependencies.add("org.spockframework:spock-spring:$spockVersion")
+        }
+
+        project.logger.lifecycle("Applying spock support")
+
+        with(project) {
+            with(plugins) {
+                apply("groovy")
+
+                withId("org.jetbrains.kotlin.jvm") {
+                    withGroovyBuilder {
+                        "compileTestGroovy" {
+                            "dependsOn" to "compileTestKotlin"
+                        }
+                        "compileTestGroovy.classpath += files(compileTestKotlin.destinationDir)"
```

Comment:
w: /Users/m89870/dev/test/aurora-gradle-plugin/src/main/kotlin/no/skatteetaten/aurora/gradle/plugins/mutators/JavaApplicationTools.kt: (367, 25): The expression is unused sier mitt bygg her.